css/main.css

```diff
@@ -16,68 +11,144 @@ pre {
 ul, ol {
     margin-top: 1.5em;
 }
-h1, h2 {
+h1 {
+    margin-top: 1.5em;
+    font-size: 2.3em;
+}
+h2 {
     margin-top: 1.5em;
+    font-size: 2em;
 }
 ul, ol {
     margin-left: 1em;
 }
 
 h1, h2, h3 { font-weight: 300; }
 
+a:link, a:visited {
+    color: #1c6fef;
+}
+
+html {
+    min-height: 100%;
+    margin: 0;
+}
+
 body {
     display: flex;
     flex-direction: column;
     align-items: center;
     min-height: 100vh;
     font-family: "Open Sans", "DejaVu Sans", sans-serif;
+    font-size: 20px;
+    line-height: 1.5em;
+    margin: 0;
 }
 
 header {
+    width: 100%;
+    box-shadow: 0 0 7px rgba(0, 0, 0, 0.15);
+}
+
+#header-inner {
     display: flex;
+    flex-wrap: wrap;
     justify-content: space-between;
-    align-items: baseline;
+    align-items: stretch;
     width: 100%;
     max-width: 80ch;
     padding: 1ch;
-    border-bottom: 1px solid rgba(0, 0, 0, .8);
+    margin: 0 auto;
 }
 
-.nav-link {
+#header-logo {
+    display: block;
+    width: 200px;
+    margin-right: 7px;
+}
+
+#header-logo:hover, #header-logo:focus {
+    opacity: 0.8;
+}
+
+a.nav-link, a.site-title {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    transition: .3s;
+}
+
+a.nav-link {
+    min-height: 40px;
+    padding: 0 20px;
 	font-family: "Open Sans", "DejaVu Sans", sans-serif;
 	font-weight: 400;
 	font-style: normal;
 	color: rgba(0, 0, 0, .8);
     text-decoration: none;
+    border-radius: 4px;
 }
 
-.nav-link:hover {
-    color: rgba(0, 0, 0, .8);
-    text-decoration: underline;
+a.nav-link:hover {
+    background-color: #eeeeee;
+}
+
+#logo {
+    width: calc(20% + 250px)
+}
+
+@media (max-width: 600px) {
```

Comment:
If possible, I'd love to keep things simple, avoid media queries and just use single layout that always works.

Reply:
I got rid of the media queries. Instead, I'm using `calc()` to adjust the font size to the screen size (I want ~15px on a smartphone and ~20px on a computer screen).